students/nskvarch/Lesson6/mailroom4.py

```diff
@@ -0,0 +1,153 @@
+        donor_name = add_new_donor(data_base, response, amount)
+        print_thankyou(donor_name)
+        return data_base
+
+
+def print_thankyou(donor_name):
+    """Takes in the name and donation amount and organizes the values into a sample thank you letter and displays
+    it on the screen """
+    nl = "\n"
+    print(f"Dear {donor_name},{nl}" +
+          f"    Thank you for your donation of $ {donor_db[donor_name][-1]}. We {nl}" +
+          f"appreciate your contribution.{nl}{nl}    Your total donation amount is now " +
+          f"$ {sum(donor_db[donor_name])}.{nl}{nl}" +
+          f"Sincerely,{nl}" +
+          f"Your Charity of Choice"
+          )
+
+
+def create_thankyou_all():
+    """Takes in the name and donation amount for all donors and creates a thank you text document
+    in the same folder the program is run from for each donor, unless a specific directory is specified"""
+    nl = "\n"
+
+    custom = input("Press Enter to create the letters in the directory the Mailroom has been run from"
+                   " or type 'custom' to specify a directory")
+    # I had to get creative here as the error catch needs to come before
+    # the Open operator so I can request input again rather than a try loop around the 'with open'
+    if custom != "":
+        while True:
+            customdir = input("Please specify a custom directory path:  ")
+            tested = os.path.exists(customdir)
+            if tested == True:
+                break
+            elif tested == False:
+                print("That directory or path does not exist, please specify a valid directory")
+        cwd = customdir
+
+    elif custom == "":
+        cwd = os.getcwd()
+    for donor_name in donor_db:
+        filename = donor_name.replace(" ", "") + ".txt"
+        full_filename = os.path.join(cwd, filename)
+        with open(full_filename, 'w') as f:
+            f.write(f"Dear {donor_name},{nl}" +
+                    f"    Thank you for your donation of $ {donor_db[donor_name][-1]}. We {nl}" +
+                    f"appreciate your contribution.{nl}{nl}    Your total donation amount is now " +
+                    f"$ {sum(donor_db[donor_name])}.{nl}{nl}" +
+                    f"Sincerely,{nl}" +
+                    f"Your Charity of Choice"
+                    )
+
+
+def add_new_donor(data_base, name, amount):
+    """add a donor not in the donor database to the database"""
+    data_base.update({name: [amount]})
+    return name
+
+
+def exit_program():
+    """use the sys module to clean-exit the script"""
+    print("Good bye")
+    sys.exit()
+
+
+def main():
+    """"main loop for the program to occupy while running, includes the main menu"""
+    while True:
+        response = input(menu)
+        switch_dict = {"1": lambda db=donor_db: thanks(db), "2": run_report, "3": create_thankyou_all}
+        if response in switch_dict:
+            switch_dict[response]()
+        elif response == "4":
+            exit_program()
```

Comment:
this shouldn't be a special case here - map exit_program() function in the switch dict.